### DISCUSSION TITLE
How do I set path parameter in camelCase to match snake_case in handler parameter?

### DISCUSSION THREAD

**Discussion by pikhovkin** [litestar-org/litestar#3846] [Q&A]
How do I set path parameter in camelCase to match snake_case in handler parameter?

For example
```python
...

@get("/user/{userId:int}")
def get_user(user_id: int) -> User:
    print(user_id)
    ...
```

**Comment by provinzkraut** [comment]
You can't. Why would you want to do that? The internally used parameter name doesn't really affect anything.

**Reply by pikhovkin** [reply]
I don't want to manually track the parameter names so that they match the parameters from the URL.
Using lowerCamelCase in handler's parameter is not pythonic way.
The internally used parameter name affects on view of API in swagger.
it's very strange to use a lowerCamelCase in content data, and then use an snake_case in URL path.

**Comment by TeeSeal** [comment]
Looking to do this as well in order to have a consistent looking OpenAPI spec.

I've been looking into ways to achieve this with OpenAPI config plugins or `type_encoders` but didn't crack it yet.

A property in the `Parameter` class that allows mapping the name of the path parameter similarly to how we can for query and header parameters would have been great, but it seems there is no such thing 😄

**Reply by pikhovkin** [reply]
I had to do it this way
```python
...

@get("/user/{userId:int}")
def get_user(userId: int) -> User:
    print(userId)
    ...
```
although this breaks the pythonic way
